### PR TITLE
selectWoo beta 2

### PR DESCRIPTION
**Pull request by claudiulodro**
Relates to #16003

This should fix the issues in VoiceOver @Umimarina reported. (I can update the JS on the accessibility testing pages if you want to give it another shot, @Umimarina. Let me know if you do.)

selectWoo has to manually manage the browser focus of the dropdowns in order to work properly on screen readers. select2 just kept the focus in the text input the whole time.

This PR also fixes some "role" issues with the nested selects. 

@mikejolley you should be able to give the 3.2 beta or master after this is merged to users waiting for this. I think it's at a good point now.

To test:
1. Bust the asset cache for selectWoo
2. Turn on VoiceOver
3. Navigate selectWoo elements (settings page and checkout page) using the keyboard.

### CODE REVIEW

File: assets/js/selectWoo/selectWoo.js [hunk 1]
```diff
@@ -5426,17 +5436,35 @@ S2.define('select2/core',[
...
         }
-      } else {
+
+        // Move the focus to the selected element on keyboard navigation.
+        // Required for screen readers to work properly.
+        if (key === KEYS.DOWN || key === KEYS.UP) {
+            self.focusOnActiveElement();
+        } else {
+          // Focus on the search if user starts typing.
+          $('.select2-search__field').focus();
```
Comment (claudiulodro) [inline assets/js/selectWoo/selectWoo.js]: Need to prevent this from focusing on other search fields if multiple selects are on the page. Small tweak.

**Comment by Umimarina** [comment]
Hello @claudiulodro, yes I can give it another shot. Where should I test the code with the correction? The URL that I have is the following: https://woocommerce.github.io/selectWoo/accessibility-testing/
Is this URL OK?

**Comment by claudiulodro** [comment]
@Umimarina Excellent! Thanks! That's the correct URL. I have to update the samples to use the latest code. I'll have that done tomorrow morning then it should be ready to test.

**Comment by claudiulodro** [comment]
@Umimarina I've updated https://woocommerce.github.io/selectWoo/accessibility-testing/ to use the latest version of selectWoo. [Here is the issue page](https://github.com/woocommerce/woocommerce/issues/16003) if you find any new issues. Thanks again!

**Pull request merged**